editor/translations/extract.py

```diff
@@ -52,11 +52,7 @@
 "Content-Transfer-Encoding: 8-bit\\n"
 """
 
-print("Updating the editor.pot template...")
-
-for fname in matches:
-
-    f = open(fname, "rb")
+def process_file(f, fname):
```

Comment:
I'm not sure why but the changes in this file broke it:
```
$ python2 editor/translations/extract.py 
Updating the editor.pot template...
Traceback (most recent call last):
  File "editor/translations/extract.py", line 104, in <module>
    process_file(f, fname)
  File "editor/translations/extract.py", line 84, in process_file
    main_po += "\n#: " + location + "\n"
UnboundLocalError: local variable 'main_po' referenced before assignment
```
If I undo those changes, it works fine:
```
$ python2 editor/translations/extract.py 
Updating the editor.pot template...
Wrapping template at 79 characters for compatibility with Weblate.
................................................................................................................................................................................................................................................................. terminé.

# Template changes compared to the staged status:
#   Additions: 571 msgids.
#   Deletions: 316 msgids.
```

Reply:
The trouble is by doing `main_po += "\n#: " + location + "\n"` you redefine `main_po`, which means `main_po` should be a local variable (i.e. defined in the function's scope)

But here `main_po` is global so you can access it fine but you get this `UnboundLocalError` if you try to redefine it.

The solution is to tell python where this variable is defined by using the `global` keyword at the begining of your function:
```python
main_po = 'godot is'

def awesomenify():
    global main_po
    main_po += 'awesome !'
```

Reply:
Ah I see, so it worked before because it was all a giant function. I'll fix that as suggested then.